### PR TITLE
Add monaco and enable column-level diffing.

### PR DESCRIPTION
**Pull request by nsthorat**
- Add monaco for viewing data like code, and diffing.
- Add a <ButtonDropdown> component for a button that opens a ComboBox since we now use it in 3 places.
- Add state to the URL

Fixes https://github.com/lilacai/lilac/issues/821

https://lilacai-nikhil-staging.hf.space/datasets#lilac/OpenOrca-100k
https://lilacai-nikhil-staging.hf.space/datasets#lilac/OpenOrca-100k&compareColumns=%5B%7B%22column%22%3A%5B%22response%22%5D%2C%22compareToColumn%22%3A%5B%22question%22%5D%2C%22swapDirection%22%3Afalse%7D%5D

https://github.com/lilacai/lilac/assets/1100749/bb810414-6e6c-4f12-84d3-189359e5fc18

### CODE REVIEW

File: web/blueprint/src/lib/view_utils.ts [hunk 1]
```diff
@@ -69,6 +69,8 @@ export const DTYPE_TO_ICON: Record<DataType, typeof CarbonIcon> = {
   null: NotAvailable
 };
 
+export const SIDEBAR_TRANSITION_TIME_MS = 250;
```
Comment (dsmilkov) [inline web/blueprint/src/lib/view_utils.ts:72]: make this is a little faster to avoid a few text re-layouts during transition. maybe 150

**Review by dsmilkov — APPROVED** [review]
awesome work! how big is the dep?

File: web/blueprint/src/lib/view_utils.ts (hunk 1 above)
Comment (nsthorat) [reply web/blueprint/src/lib/view_utils.ts]: Done!

**Comment by nsthorat** [comment]
~17K over the wire for the editor, but definitely worth it since I think we should switch everything over.

**Comment by nsthorat** [comment]
And also, it's fetched separately, so will get heavily cached.

**Pull request merged**